spec/main_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+books = ['dummy'] # add finished books to this array
+
+RSpec.describe 'books' do
+  before(:all) do
+    books.each do |book|
+      # call the bake script for each book
+      output = `USE_LOCAL_KITCHEN=1 ./bake -b #{book} -i spec/books/#{book}/input.xhtml \
+        -o spec/books/#{book}/actual_output.xhtml`
+      puts output
+    end
+  end
+
+  books.each do |book|
+    it "#{book} matches expected output" do
+      expect("spec/books/#{book}/expected_output.xhtml").to be_same_file_as(
+        "spec/books/#{book}/actual_output.xhtml"
+      )
```

Comment:
I have done things like this on occasion, so it isn't wrong.  But if you want to run one specific book's spec over and over you can't.  So I'd suggest breaking out individual examples for each book.  See my other comment above.